x/reports/keeper/external_hooks.go

```diff
@@ -53,6 +44,15 @@ func (h Hooks) AfterSubspaceDeleted(ctx sdk.Context, subspaceID uint64) {
 		h.k.DeleteReport(ctx, report.SubspaceID, report.ID)
 		return false
 	})
+
+	// Delete the reason id key
+	h.k.DeleteNextReasonID(ctx, subspaceID)
+
+	// Delete all the reasons related to this subspace
+	h.k.IterateSubspaceReasons(ctx, subspaceID, func(reason types.Reason) (stop bool) {
+		h.k.DeleteReason(ctx, reason.SubspaceID, reason.ID)
+		return false
+	})
```

Comment:
Why was this code moved here instead of keeping it above?

Reply:
`DeleteReason` also deletes reports containing reasons [here](https://github.com/desmos-labs/desmos/blob/master/x/reports/keeper/reasons.go#L83), but it causes O(N^2) since it iterates all the reports when removing a reason.

Moving this code above can have better performance with O(2N) time complexity since reports containing reasons are removed first.